### ISSUE TITLE
[Editor] Better workflow for placing warps

### ISSUE THREAD

**Issue by Emral** [WohlSoft/Moondust-Project#511]
**Is your feature request related to a problem? Please describe.**
Placing a lot of warps is tedious. Here are some suggestions. They have different niches and are all things that would be nice to have, but any of them is a step in the right direction, I think.

**Describe the solution you'd like**
1. Allow the width and height of a warp to be configured, saving it to the lvlx file format and allowing it to be parsed by all engines that want to. This solution makes placing instant/portal warps along pits faster in particular.
2. Hotkeys. Configurable keyboard shortcuts for the following: When nothing is selected, press a hotkey to put the first un-placed warp entrance or exit onto your cursor. If there is no un-placed warp entrance or exit, it creates a new warp.
3. A button in the warp settings window: "Copy", which makes a new warp with the exact same configuration and focuses it in warp settings
4. This is more of a bigger feature and should probably be its own feature request, but it would be nice to be able to ctrl/alt-drag an object or multi-selection to make a copy of it. When used on a warp, the entrance and exit should both be selected automatically so they can be moved the same way until the selection is lifted.

**Additional context**
Here are scribbles.
1.
![image](https://user-images.githubusercontent.com/4628923/185114180-cbb9e1f5-5dad-4cbf-aee7-9bd8743dc4bb.png)

2.
![image](https://user-images.githubusercontent.com/4628923/185114919-e7959f94-5c7c-47c3-b32c-9174bae1e74d.png)

3.
![image](https://user-images.githubusercontent.com/4628923/185115351-0010f378-1c18-4016-83a6-65d0aaa443bd.png)

4.
![image](https://user-images.githubusercontent.com/4628923/185115802-fee4b25b-c2ba-4eb0-b10a-93dcefe7b225.png)

**Comment by Emral** [comment]
Hm. I'm starting to think 4 might be made redundant by 1 actually. The scenario I had in mind was for placing a pit of instant warps, but with wider warps that is not necessary. And I also forgot you can ctrl ctrlv or middle mouse button a selection of stuff to clone it. The other 3 have a lot more utility for sure I think.

I like your idea of being able to switch the direction of a warp point by the context menu.

**Comment by Wohlstand** [comment]
The pit of instant warps is the only feature for users who wants make the level to run on legacy engines that don't feature any wide points at all. But, this could be useful if you want to make a long corridor of doors into different rooms.

**Comment by Alucard648** [comment]
> The pit of instant warps is the only feature for users who wants make the level to run on legacy engines that don't feature any wide points at all. But, this could be useful if you want to make a long corridor of doors into different rooms.

Indeed. Hub levels, major dungeons, Lost Woods styled mazes, all benefit from optimizing mass warp point placement.
